src/app/pages/elements/form/FormDemo.ts

```diff
@@ -338,22 +349,50 @@ export class FormDemoComponent {
         },
         state: {
           label: 'State',
+          pickerConfig: {
+            field: 'value',
+            format: '$label',
+            options: (query, countryID) => {
+              return new Promise((resolve, reject) => {
+                resolve(this.getStateOptions(query, countryID));
+              });
+            },
+            getLabels: (value: number) => {
+              return new Promise((resolve, reject) => {
+                resolve(this.getStateLabel(value));
+              });
+            }
+          }
         },
-        country: {
+        countryID: {
           label: 'Country',
+          pickerConfig: {
+            field: 'value',
+            format: '$label',
+            options: (query) => {
+              return new Promise((resolve, reject) => {
```

Comment:
return Promise.resolve(this.getCountryOptions(query));

Reply:
done